### PR TITLE
SFMU: Phase One

### PR DESCRIPTION
**Pull request by kean**
To test:

**Happy Path**

- Invoke the following universal link: https://apps.wordpress.com/get/?campaign=qr-code-media#%2Fmedia%2F<siteID> with <siteID> replaced with an actual siteID.
- **Verify** that the `deep_linked` event is send with the following properties:

```
{
    "source": "banner"
    "section": "my_site"
    "url": "/media/:domain"
    "source_info":  "qr-code-media"
}
```

- **Verify** that the Site Media screens opens and the "Choose from Device" picker is automatically invoked

https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/b1c50ac4-b5ad-4827-b44d-5905e0f070d2

**Invalid Site ID**

- Invoke the universal link with an invalid Site ID or a Site ID belonging to a different user
- **Verify** that the `deep_link_failed` event is send with the following properties:

```
{
    "error": "invalid_site_id"
    "campaign": "qr-code-media"
    "route": "/media/:domain"
}
```

- **Verify** that the link is redirected to the browser (unless you have a WP-iOS app installed then it goes into recursion...)

**Different Site**

- Invoke the universal link with a Site ID that belongs to the user but is not currently selected
- **Verify** that the app switches to a new site

## Regression Notes
1. Potential unintended areas of impact: Media, Universal Links
2. What I did to test those areas of impact (or what existing automated tests I relied on): yes
3. What automated tests I added (or what prevented me from doing so): yes

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m [hunk 1]
```diff
@@ -524,7 +529,8 @@ - (void)showDetailViewForSubsection:(BlogDetailsSubsection)section
             [self.tableView selectRowAtIndexPath:indexPath
                                         animated:NO
                                   scrollPosition:[self optimumScrollPositionForIndexPath:indexPath]];
-            [self showMediaLibraryFromSource:BlogDetailsNavigationSourceLink];
+            BOOL showPicker = userInfo[[BlogDetailsViewController userInfoShowPickerKey]] != NULL;
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m]:
```suggestion
            BOOL showPicker = (userInfo[[BlogDetailsViewController userInfoShowPickerKey]] ?: NO) && self.blog.userCanUploadMedia;
```

- I feel `userInfo[[BlogDetailsViewController userInfoShowPickerKey]] ?: NO` is more semantically correct than `userInfo[[BlogDetailsViewController userInfoShowPickerKey]] != NULL`, wdyt?
- If a user can't upload media, let's show the media screen instead

**Review by momo-ozawa — COMMENTED** [review]
We need to add a check to see if the user can upload media for the blog, but works as described otherwise!

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m (hunk 1 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m]: Addressed both comments. Nice catch with the upload eligbiility bug.

**Review by momo-ozawa — APPROVED** [review]
Thanks for addressing my comments 🙇‍♀️

**Pull request merged**